zilliqa/src/consensus.rs

```diff
@@ -2814,32 +2768,18 @@ impl Consensus {
         })
     }
 
-    /// Retrieves cached total stake for specific block number, retaining only N highest blocks.
-    fn total_weight(&mut self, executed_block: BlockHeader) -> u128 {
-        // return cached value
-        if let Some(stake) = self.stake_cache.get(&executed_block.number) {
-            return *stake;
-        }
-
-        // keep cache small
-        if self.stake_cache.len() == 10 {
-            self.stake_cache.pop_first(); // remove oldest block stake
-        }
-
-        // populate cache
-        let stake = self.total_weight_raw(executed_block);
-        self.stake_cache.insert(executed_block.number, stake);
-
-        stake
-    }
-
-    /// Retrieves total stake from Deposit contract via EVM.
-    fn total_weight_raw(&self, executed_block: BlockHeader) -> u128 {
-        self.state
-            .get_total_stake(executed_block)
-            .unwrap()
-            .unwrap()
-            .into()
+    fn total_weight(&self, committee: &[NodePublicKey], executed_block: BlockHeader) -> u128 {
+        committee
+            .iter()
```

Comment:
Just sticking this comment here as a note.
This would be O(n), though the state caching mechanism now makes this cheap.